File: src/main/java/org/eightsines/decompiled/A26.java
package org.eightsines.decompiled;

import kotlin.jvm.JvmField;

public final class A26 {
    private final boolean foo(double x, double y, double r, Point p) {
        double x$iv = x - p.x;
        double var1 = x$iv * x$iv;
        x$iv = y - p.y;
        double var2 = x$iv * x$iv;
        return var1 + var2 < r * r;
    }

    private static final class Point {
        @JvmField
        public final double x;

        @JvmField
        public double y;

        public Point(double x, double y) {
            this.x = x;
            this.y = y;
        }
    }
}

File: src/main/kotlin/org/eightsines/kotlinperfopt/A26.kt
package org.eightsines.kotlinperfopt

@Suppress("NOTHING_TO_INLINE")
private inline fun sq(x: Double) = x * x

class A26 {
    private class Point(@JvmField val x: Double, @JvmField var y: Double)

    private fun foo(x: Double, y: Double, r: Double, p: Point) =
        sq(x - p.x) + sq(y - p.y) < sq(r)
}
